src/main/java/schwarz/jobs/interview/coupon/web/dto/CouponDTO.java

```diff
@@ -5,12 +5,17 @@
 import lombok.Builder;
 import lombok.Data;
 
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
 @Data
 @Builder
 public class CouponDTO {
 
+    @NotNull(message = "The discount cannot be null.")
```

Comment:
Added annotations to align with database constraints and to enable validation of the object, ensuring data integrity and correctness before processing.